docs/3.0.0-beta.x/migration-guide/migration-guide-alpha.26-to-beta.md

```diff
@@ -0,0 +1,736 @@
+  },
+
+  create: async (ctx) => {
+    return strapi.services.product.add(ctx.request.body);
+  },
+
+  update: async (ctx) => {
+    return strapi.services.product.edit(ctx.params, ctx.request.body) ;
+  },
+
+  destroy: async (ctx) => {
+    return strapi.services.product.remove(ctx.params);
+  }
+};
+
+```
+
+**After**
+
+`./api/product/controllers/Product.js`
+
+```js
+module.exports = {
+  // this function overrides the default one
+  find: async ctx => {
+    myCustomImplementation(ctx);
+  },
+
+  // this one is added to the default controller
+  customAction: async ctx => {
+    ctx.send({ ok: true });
+  },
+};
+```
+
+> The same goes for services
+
+For custom `controllers` and `services` (the ones without a model) you can leave them untouched.
+
+### Migrating routes
+
+In the file `./api/apiName/config/routes.json` we renamed the `destroy` action to `delete` you will have to update your `routes` and `controller` accordingly.
+
+If you haven't customized the `destroy` action then remove it from your controller instead of renaming it.
+
+**Before**
+
+`./api/product/config/routes.json`
+
+```json
+{
+  "routes": [
+    //...
+    {
+      "method": "DELETE",
+      "path": "/products/:id",
+      "handler": "Product.destroy",
+      "config": {
+        "policies": []
+      }
+    }
+  ]
+}
+```
+
+`./api/product/controllers/Product.js`
+
+```js
+module.exports = {
+  find: async ctx => {
+    return strapi.services.product.fetchAll(ctx.query);
+  },
+
+  findOne: async ctx => {
+    return strapi.services.product.fetch(ctx.params);
+  },
+
+  count: async ctx => {
+    return strapi.services.product.count(ctx.query);
+  },
+
+  create: async ctx => {
+    return strapi.services.product.add(ctx.request.body);
+  },
+
+  update: async ctx => {
+    return strapi.services.product.edit(ctx.params, ctx.request.body);
+  },
+
+  destroy: async ctx => {
+    customImplementation(ctx);
+  },
+};
+```
+
+**After**
+
+`./api/product/config/routes.json`
+
+```json
+{
+  "routes": [
+    //...
+    {
+      "method": "DELETE",
+      "path": "/products/:id",
+      "handler": "Product.delete",
+      "config": {
+        "policies": []
+      }
+    }
+  ]
+}
+```
+
+`./api/product/controllers/Product.js`
+
+```js
+module.exports = {
+  delete: async ctx => {
+    customImplementation(ctx);
+  },
+};
+```
+
+## Migrating `admin`
+
+Customizing the admin is as simple as creating a file in the `./admin` folder of your app. You need to make sure the file you want to customize is at the same location in your `./admin` folder as it is in the `strapi-admin` package. For a reference you can look at the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin).
+
+For the beta there are quite a lot of changes made to the admin. If you did customize things you will have to verifiy if the file still exists in the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin) or find its new location.
```

Comment:
If you **previously customized** things you will have to **verify** if the file still exists in the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin) or find its new location.